src/main/java/org/spongepowered/api/text/TextArguments.java

```diff
@@ -0,0 +1,141 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.text;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+
+/**
+ * A utility class to generate useful {@link TextArgument}s that can be passed
+ * to {@link Text.Placeholder}.
+ */
+public final class TextArguments {
+
+    /**
+     * Creates a new function that will return the value for the given key.
+     *
+     * @param key The key used to fetch the data
+     * @return The newly created TextArgument
+     */
+    public static TextArgument valueForKey(String key) {
+        return new KeyFunction(key);
+    }
+
+    /**
+     * Creates a new function that will transform the value for the key with the
+     * given function in case its present.
+     *
+     * @param textArgument The textArgument used to get the data
+     * @param function The transform function used to transform the data
+     *        returned from function
+     * @return The newly created TextArgument
+     */
+    public static <F> TextArgument transform(TextArgument textArgument, Function<F, ?> function) {
+        return new Transform<F>(textArgument, function);
+    }
+
+    // /**
+    //  * Creates a new function that will try to get the valueKey from the
+    //  * {@link DataHolder} provided for the given key.
+    //  *
+    //  * @param key The key used to fetch the data
+    //  * @param function The function used to transform the data
+    //  * @return The newly created TextArgument
+    //  */
+    // public static TextArgument transform(String key, Key<? extends BaseValue<?>> valueKey) {
+    //     return new Transform<DataHolder>(valueForKey(key), input -> input.get(valueKey));
+    // }
+
+    static final class Transform<F> implements TextArgument {
+
+        private final TextArgument textArgument;
+        private final Function<Object, ?> function;
+
+        @SuppressWarnings("unchecked")
+        Transform(TextArgument textArgument, Function<F, ?> function) {
+            this.textArgument = checkNotNull(textArgument, "textArgument");
+            this.function = (Function<Object, ?>) checkNotNull(function, "function");
+        }
+
+        @Override
+        public Optional<?> apply(Function<String, ?> input) {
+            return this.textArgument.apply(input).transform(this.function);
+        }
+
+    }
+
+    /**
+     * Creates a new function that will return an empty text in case the wrapped
+     * TextArgument returns null.
```

Comment:
In Sponge, there's a very commonly-held convention to use Optional for public-facing interfaces. How could TextArgument return null, especially when it returns Optional?